lib/serviceDiscovery/engines/consul/index.js

```diff
@@ -89,20 +89,19 @@ class ConsulService extends Service {
 	announce(port, metadata, cb) {
 		logger.debug(`Announcing service ${this.name} on port ${port}`);
 		this._announces.push({port, metadata});
-		this.session((err, id) => {
+		this.session((err) => {
 			if (err) {
 				return cb(err);
 			}
 
-			const key = [this.path, os.hostname(), process.pid].join('/');
+			const key = [this.path, os.hostname(), helpers.generateId(port)].join('/');
```

Comment:
Should we remove the hostname? In dockerized environment, the hostname will be the container ID, and a new container will have a different ID but might have the same generated ID.

Reply:
That's a good point.  I'll switch it to `this.path, helpers.generateId` instead.